### DISCUSSION TITLE
Compactness penalty

### DISCUSSION THREAD

**Discussion by kmytsay** [JeschkeLab/DeerLab#381] [General] [answered]
I have a similar question about the compactness penalty. It seems that the Non-linear least-squares routine runs until an optimized compactness parameter is reached. This process is quite slow, is it possible for the fitting routine to fit all compactness parameter at the same time?

Do you recommend using bootstrapping without compactness penalty for speed?

**Comment by luisfabib** [comment, marked as answer]
Sorry for the delay (had a thesis to submit :P). I hope the answer still helps you!

I would recommend to run first the analysis with compactness to get the optimal value of the compactness regularization parameter and then run the bootstrapping with compactness using the optimal value fixed. This should keep your analysis quick but robust. 

For example:
```python 
# Run first run to optimize the compactness via the ICC
results_ = dl.fit(Vmodel,Vexp, penalties=compactness)
compactness_weightopt = results_.penweights

# Run bootstrapped analysis with fixed compactness penalty weight
compactness.weight.freeze(compactness_weightopt)
results = dl.fit(Vmodel,Vexp, penalties=compactness, bootstrap=100)
```

**Reply by kmytsay** [reply]
Thank you, Luis! Good luck on your thesis defense :)
